polaris-react/src/types.ts

```diff
@@ -34,10 +34,7 @@ export interface SectionDescriptor {
 
 export type Descriptor = SectionDescriptor | OptionDescriptor;
 
-export type IconSource =
-  | React.FunctionComponent<React.SVGProps<SVGSVGElement>>
-  | 'placeholder'
-  | string;
+export type IconSource = React.ComponentType<React.SVGProps<SVGSVGElement>>;
```

Comment:
I didn't see any constraints in our `Icon` implementation that requires the `source` to be a function component. Therefore, I updated the type from `React.FunctionComponent` to `React.ComponentType` (allowing both class and function components)